tests/test_scenarios.py

```diff
@@ -0,0 +1,173 @@
+        clen = int(req.headers.get("content-length"))
+        assert clen > 0
+        log.debug(f"{clen=}")
+        log.debug(f"{ctype=}")
+
+        g = Graph().parse(url)
+        # ttl = g.serialize(format="turtle").strip()
+        # log.debug(f"{ttl=}")
+        log.debug(f"{len(g)=}")
+
+
+# launch a server with subprocess from local_server.py
+@pytest.mark.usefixtures("httpd_server_base", "store_info_sets")
+def test_scenarios(
+    httpd_server_base: str,
+    store_info_sets,
+):
+    assert httpd_server_base
+
+    for config in Path(CONFIGS).glob("*.yml"):
+        if config.name not in SCENARIOS_OUTCOMES:
+            log.debug(f"Skipping scenario {config}")
+            continue
+        for test_case in SCENARIOS_OUTCOMES[config.name]:
+            mode = test_case["mode"]
+            log.debug(f"{mode=}")
+
+            # TODO: when mode is added in 0.0.5 add tests here
+            # for now continue
+
+            if mode == "nostop":
+                log.debug(f"Skipping scenario {config} with mode {mode}")
+                continue
+
+            for store_info in store_info_sets:
+                log.debug(f"Running scenario {config} with store {store_info}")
+                travharv = TravHarv(
+                    config,
+                    store_info,
+                )
+                travharv.process()
+                # assert travharv.error_occurred
+                # other assertions here based on the scenarios
+
+                # make connection with the store
+                # create RDF_store
+                core_store: RDFStore = create_rdf_store(*store_info)
+                rdf_store_access = RDFStoreAccess(core_store)
+
+                # get the context name for the config name
+                context_name_graph = rdf_store_access._nmapper.key_to_ng(
+                    config.name
+                )
+                log.debug(f"{context_name_graph=}")
+
+                # make sparql for the test
+                sparql = QUERY_BUILDER.build_syntax(
+                    "execution_report_data.sparql",
+                    context_name_graph=context_name_graph,
+                )
+
+                log.debug(f"{store_info=}")
+
+                # if store info is an empty tuple then skip
+                if store_info == ():
+                    sparql = QUERY_BUILDER.build_syntax(
+                        "execution_report_data.sparql",
+                    )
+                    execution_report_data_result = (
+                        travharv.target_store.select(sparql)
+                    )
+                    sparql_all = "SELECT ?s ?p ?o WHERE { ?s ?p ?o }"
+                    all_triples = travharv.target_store.select(sparql_all)
+                else:
+                    execution_report_data_result = rdf_store_access.select(
+                        sparql
+                    )
+                    sparql_all = (
+                        "SELECT ?s ?p ?o FROM <"
```

Comment:
why not use f string?  do watch out to escape the `{}` think it is done via `where {{ ?s ?p ?o .}}`